text/0309-parameter-cross-stack-ref.md

```diff
@@ -0,0 +1,87 @@
+---
+rfc pr: [#311](https://github.com/aws/aws-cdk-rfcs/pull/311)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/309
+---
+
+# [Loosely Coupled Cross Stack Ref]
+
+Use Parameter Store parameters for cross stack references
+
+## Working Backwards
+
+When passing object or class properties between stacks, instead of creating a CFN Export,
+Create a SSM Parameter in parameter store, then have the consuming stack create a parameter
+to consume the value from parameter store
+
+### CHANGELOG
+
+feat(core): loosely coupled cross stack references
+
+BREAKING CHANGE: Stacks will need to decouple their exports with dummy values
+
+### README
+
+#### Loosely Coupled Stack References
```

Comment:
I think I'd prefer the term "**weak** cross-stack references"

Reply:
@rix0rrr https://www.endoflineblog.com/cdk-tips-03-how-to-unblock-cross-stack-references describes the workaround necessary to delete a resource that's exported from one stack and consumed by another using the tightly-coupled approach. Moving to loosely-coupled puts the responsibility on the shoulders of the user, which, I think is reasonable, or at least it's an option I'd use if it were available.

I can't think of any case where the strong guarantees of export are enough of a benefit that I'd actually pick them over params. I would absolutely LOVE to be _able_ to delete a stack which other stacks depend on. Because while it may sound like the wrong decisions and may even be the wrong decision most of the time, there are times when I could imagine it would be the best of a bunch of bad options.

I really like your observation about `both` mode though. I thing that the export model really needs to find a way to "still produce the required exports" in the case where resources are being deleted whether they're being replaced by params or just refactored out of the code in general.